### PR TITLE
Create changes file

### PR DESCRIPTION
**Pull request by Jessie212**

### CODE REVIEW

**Comment by Jessie212** [comment]
@ebyhr What would you like me to do about the previous versions? Should we just mention that features from all the releases are documented in the README?

File: CHANGES.md
Comment (mosabua) [inline CHANGES.md:55]: Remove link to PRs here

File: CHANGES.md
Comment (mosabua) [inline CHANGES.md:57]: Add a note about older version here

File: CHANGES.md
Comment (mosabua) [inline CHANGES.md:42]: empty line above list

**Review by mosabua — APPROVED** [review]
Minor add but overall good to go.

**Review by hashhar — CHANGES_REQUESTED** [review]
I don't think there's a process proposed here so just adding this file won't help with maintaining release notes.

There's no point in listing past releases. It's not useful use of time to vet what changes went in past releases.

One alternative is something like in https://github.com/trinodb/charts/releases/tag/trino-0.8.0 where anything not explicitly labelled gets added to release notes automatically. It can then be later hand-edited as needed.

**Comment by mosabua** [comment]
We are adding this file as suggested by you since there was no other release notes list. We pulled the info from the announcements on slack. The proposed process changes is to just update this file prior to each release manually. Then each tag has the latest info in it and master branch also has it readily available. We can then link to this document from other sources that want to refer to specific releases or the changes in general.

**Comment by hovaesco** [comment]
FYI we recently introduced [changie](https://changie.dev/) in dbt-trino https://github.com/starburstdata/dbt-trino/pull/98 (and other useful stuff like linting, formatting etc.). It automates creating of [CHANGELOG](https://github.com/starburstdata/dbt-trino/blob/master/CHANGELOG.md).

**Comment by leniartek** [comment]
@mosabua Going forward we will automate change log creation as we did in dbt-trino https://github.com/starburstdata/dbt-trino/blob/master/CHANGELOG.md

I don't mind adding the past release notes manually if you want to.

Screenshot 2022-08-18 at 14 01 16

**Comment by mosabua** [comment]
okay ... then we can close this PR maybe? Either way .. just let us know when that page is up..

**Comment by hashhar** [comment]
@leniartek That release notes page has some issues - the people who contribute changes shouldn't be the one contributing release notes. It needs an editorial process. As an example none of the "features" mentioned in the release notes mention how to opt into or use that feature. Same for bugs - they don't say when or how could it be encountered.

We should involve @mosabua / @colebow for the initial process and leverage their help to see what parts can and cannot be automated.

**Comment by mosabua** [comment]
Is there any progress on how you want to proceed @leniartek @hovaesco @hashhar and others?

**Comment by hashhar** [comment]
@mosabua I'm in favour of the approach proposed here because the people writing release notes should not be same as the one who added the feature since it leads to pointless changelogs (see the example in my comment above).

**Comment by hashhar** [comment]
I'll merge this PR but I still think we should not document past releases since those release notes are not complete and only a highlight of select features from a release.

**Comment by hovaesco** [comment]
I would argue that it leads to pointless changelogs since it comes within PRs, so maintainer can review it, suggest rewording, propose something different etc.   

I'm fine with whatever approach which will improve the quality of delivering new releases. We all see a need of having release notes, so let's just figure out who will be responsible for maintaining them and what process we can follow.

**Comment by hovaesco** [comment]
I've opened a PR which adds a pull request template, since we should follow the same process as in Trino project. https://github.com/trinodb/trino-python-client/pull/234

**Comment by mosabua** [comment]
So I think we need to decide how to proceed. If we follow the same approach as in trino, then we would merge the PR template, merge this PR, and then create a PR that updates the info for the next release. Then keep updating that PR and merge it just before release.

We are happy to help with this implementation just like we do in Trino .. if you are happy to go down that route.

One thing we would have to decide is if we use this current content and just get better... or if we start with zero info. Personally I think some release notes (like in this PR) are better than none .. but YMMV

**Comment by hashhar** [comment]
> So I think we need to decide how to proceed. If we follow the same approach as in trino, then we would merge the PR template, merge this PR, and then create a PR that updates the info for the next release. Then keep updating that PR and merge it just before release.

Yes. That makes sense. We want to start a process similar to Trino for now and we can see what can be improved over time.

> We are happy to help with this implementation just like we do in Trino .. if you are happy to go down that route.

Thank you.

> One thing we would have to decide is if we use this current content and just get better... or if we start with zero info. Personally I think some release notes (like in this PR) are better than none .. but YMMV

I tend to think that starting with no information is better since the current release notes are just some selected highlights and slightly misleading (since they are not complete). But I'm with either.

**Review by hashhar — APPROVED** [review]

**Comment by colebow** [comment]
> Ping @colebow to acknowledge since we'd need his help as well. I'll merge this then.

Happy to help in whatever way is needed.

**Comment by hashhar** [comment]
Thanks @colebow. I'm merging this PR - the process is similar to what we do in Trino.

- Each PR (at-least recently opened ones) will have a release notes in PR template.
- We keep a PR open with draft release notes - in CHANGES.md added in this PR
- Before a release is cut we review and merge CHANGES.md
- Perform a release and announce.

File: CHANGES.md
Comment (hashhar) [inline CHANGES.md:16]: I added these for 0.316.0.

Comment (hashhar) [reply CHANGES.md:16]: @colebow can you review this please.

File: CHANGES.md
Comment (hovaesco) [inline CHANGES.md:11]: Does `#186` link to a PR or it misses a link?

Comment (hashhar) [reply CHANGES.md:11]: good point. It looks like the magic linking only works in PR/issue comments and commit messages.

I'll fix this.

Comment (hashhar) [reply CHANGES.md]: Looks like this now. Thanks for catching.
image

File: CHANGES.md
Comment (colebow) [inline CHANGES.md:17]:
```suggestion
* Fix incorrect results for `get_table_comment` in SQLAlchemy when two tables
  with the same name and schema exist in different catalogs.
```

File: CHANGES.md
Comment (colebow) [inline CHANGES.md:19]:
```suggestion
* Remove spurious logging of HTTP responses when a query is cancelled.
```

**Review by colebow — APPROVED** [review]
Two small nits, otherwise LGTM.

**Comment by mosabua** [comment]
Can we get this merged and then do an update to the new release in a follow up PR? @Jessie212 is unavailable this week

**Comment by hashhar** [comment]
Thanks @Jessie212 and others for contributing this and helping make sure future versions come with release notes.

**Pull request merged**
